docs/reference/components/sources/exec.cue

```diff
@@ -0,0 +1,306 @@
+			common:        false
+			description:   "Determine if events should be generated per line."
+			relevant_when: "mode = `scheduled`"
+			required:      false
+			type: bool: default: true
+		}
+		exec_duration_millis_key: {
+			category:      "Context"
+			common:        false
+			description:   "The key name added to each event representing the duration in millis that the command took to run."
+			relevant_when: "mode = `scheduled`"
+			required:      false
+			warnings: []
+			type: string: {
+				default: "exec_duration_millis"
+				syntax:  "literal"
+			}
+		}
+		respawn_on_exit: {
+			common:        false
+			description:   "Determine if a streaming command should be restarted if it exits."
+			relevant_when: "mode = `streaming`"
+			required:      false
+			type: bool: default: true
+		}
+		respawn_interval_secs: {
+			common:        false
+			description:   "The interval in seconds between restarting streaming commands if needed."
+			relevant_when: "mode = `streaming`"
+			required:      false
+			warnings: []
+			type: uint: {
+				default: 60
+				unit:    "seconds"
+			}
+		}
+		current_dir: {
+			common:      false
+			required:    false
+			description: "The directory from within which to run the command."
+			warnings: []
+			type: string: {
+				default: null
+				syntax:  "literal"
+			}
+		}
+		include_stderr: {
+			common:      false
+			description: "Include the output of stderr when generating events."
+			required:    false
+			type: bool: default: false
+		}
+		host_key: {
+			category:    "Context"
+			common:      false
+			description: "The key name added to each event representing the current host. This can also be globally set via the [global `host_key` option][docs.reference.configuration.global-options#host_key]."
+			required:    false
+			warnings: []
+			type: string: {
+				default: "host"
+				syntax:  "literal"
+			}
+		}
+
+		pid_key: {
+			category:    "Context"
+			common:      false
+			description: "The key name added to each event representing the process ID of the running command."
+			required:    false
+			warnings: []
+			type: string: {
+				default: "pid"
+				syntax:  "literal"
+			}
+		}
+		exit_status_key: {
+			category:    "Context"
+			common:      false
+			description: "The key name added to each event representing the exit status of a scheduled command."
+			required:    false
+			warnings: []
+			type: string: {
+				default: "exit_status"
+				syntax:  "literal"
+			}
+		}
+		command_key: {
+			category:    "Context"
+			common:      false
+			description: "The key name added to each event representing the command that was run."
+			required:    false
+			warnings: []
+			type: string: {
+				default: null
+				syntax:  "literal"
+			}
+		}
+		argument_key: {
+			category:    "Context"
+			common:      false
+			description: "The key name added to each event representing the arguments that were provided to the command."
+			required:    false
+			warnings: []
+			type: string: {
```

Comment:
Should also be `[string]`

Reply:
Apologies but I think I missed one and now am confused. For the following should the type of the items in the array be string or [string]:-

>         arguments: {
>             common:      false
>             description: "Array of any arguments to pass to the command."
>             required:    false
>             type: array: {
>                 default: null
>                 items: type: [string]: {
>                     examples: ["Hello World!", "-la"]
>                     syntax: "literal"
>                 }
>             }
>         }

i.e. 
>                 items: type: [string]: {

or

>                 items: type: string: {

Sorry for the confusion!

Reply:
I think it should be `[string]` there and then the examples would become:

```
examples: [["Hello World"], ["-l", "-a"]]
```

Let me know if cue doesn't like that though.

Reply:
Hmm, actually, I'm wrong! I think what you have there is ok. I verified by looking at the `lua` cue which has a similar array argument.

Reply:
OK, I think the new documentation is correct now.